### ISSUE TITLE
PubSubClient does not catch all messages send after a subscribe

### ISSUE THREAD

**Issue by stefanbode** [knolleary/pubsubclient#325]
This might be an ESP8266 only problem but I can confirm, that if you subscribe to a topic and there is more than ONE retained message at the broker, the callback() function only delivers the first message sent by the broker. All other messages get lost in nirvana.

**Comment by Cosford** [comment]
I am fairly certain this is correct behaviour on a per topic basis. For each topic, the broker stores only one retained message, which is correctly delivered to new subscribers.

**Comment by stefanbode** [comment]
You are correct if you subscribe on ONE topic without wildcards. If you use the # or the + there can be more than one topic behind this subscription. This will cause the issue.
